frontend/components/ActivityDialog.vue

```diff
@@ -66,6 +69,72 @@ interface ActivityForm {
 
 type ActivityFormErrors = Partial<Record<keyof ActivityForm, string>>;
 
+const confirmDelete = () => {
+    confirm.require({
+        header: t.value("activity.delete.header"),
+        message: t.value("activity.delete.confirm"),
+        icon: "pi pi-exclamation-triangle",
+        rejectClass: "hover:underline",
+        acceptClass:
+            "text-error dark:text-error-dark hover:underline font-bold",
+        rejectLabel: t.value("common.button.cancel"),
+        acceptLabel: t.value("journey.delete"),
+
+        accept: () => {
+            toast.add({
+                severity: "info",
+                summary: t.value("common.toast.info.heading"),
+                detail: t.value("delete.journey.toast.message"),
+                life: 3000,
+            });
+            deleteActivity();
+        },
+    });
+};
+
+async function deleteActivity() {
+    await client(`/api/journey/${props.id}/activity/${props.activityId}`, {
+        method: "delete",
+        body: {},
+        async onResponse({ response }) {
+            if (response.ok) {
+                toast.add({
+                    severity: "success",
+                    summary: t.value(
+                        "form.input.activity.delete.toast.success.heading",
+                    ),
+                    detail: t.value(
+                        "form.input.activity.delete.toast.success.detail",
+                    ),
+                    life: 6000,
+                });
+                close();
+                const { data: activityData } = await useAsyncData(
```

Comment:
Theoretically the activity could just be removed in the front end without a fetch call